conda_env_tracker/cli.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+
+# conda execute
+# env:
+#  - gitpython
+#  - conda-execute
+#  - yaml
+# channels:
+#  - minadyn
+#  - conda-forge
+
+from __future__ import print_function
+
+import datetime
+import contextlib
+import logging
+import os
+import shutil
+import tempfile
+
+import conda.resolve
+import conda.api
+import conda_execute.config
+from git import Repo
+import yaml
+
+
+def env_check(repo_dir):
+    fname = os.path.join(repo_dir, 'env.spec')
+    if not os.path.exists(fname):
+        raise IOError("File {} doesn't exist.".format(fname))
+    with open(fname) as fh:
+        spec = yaml.safe_load(fh)
+    env_spec = spec.get('env', [])
+    index = conda.api.get_index(spec.get('channels', []))
+    r = conda.resolve.Resolve(index)
+    full_list_of_packages = sorted(r.solve(env_spec))
+    return index, spec, full_list_of_packages
+
+
+def build_manifest_branches(repo_directory):
+    r = Repo(repo_directory)
+
+    for remote in r.remotes:
+        remote.fetch()
+
+    for env in r.branches:
+        name = env.name
+        if 'manifest/' in name:
+            continue
+        env.checkout()
+        spec_fname = os.path.join(repo_directory, 'env.spec')
+        if not os.path.exists(spec_fname):
+            # Skip branches which don't have a spec.
+            continue
+        index, spec, pkgs = env_check(repo_directory)
```

Comment:
`env_check` could do with a better name ... it makes it sound like a validation check, but actually it resolves the environment spec to a list of packages.